var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -342,7 +345,8 @@ class Llvm(CMakePackage, CudaPackage, LlvmDetection, CompilerPackage):
         depends_on("libedit")
         depends_on("libxml2")
         depends_on("lua@5.3", when="+lua")  # purposefully not a range
-        depends_on("ncurses")
+        depends_on("ncurses~termlib", when="@19:")
```

Comment:
Does it require that ncurses not provide termlib now for some reason?

Reply:
I think the answer to your question is essentially "yes"

@prckent has a ref to llvm discussion [with further URLs to related discussions].

And I see @haampie has been dealing with this issue for quiet some time.

`<snip>/binutils-2.43.1-ezlfif3q225pdwdgywwod42emyguhpo5/bin/ld: IOHandlerCursesGUI.cpp:(.text._ZN12lldb_private18IOHandlerCursesGUI8ActivateEv+0x1622): undefined reference to 'define_key'`

i.e llvm has code that calls `define_key()`

ncurses+termlib has:
```
$ nm -Ao libcurses.so libtinfo.so |grep define_key
libtinfo.so:000000000002bee0 T define_key
libtinfo.so:000000000002bdc0 T define_key_sp
```
ncurses~termlib has:
```
$ nm -Ao libcurses.so libtinfo.so |grep define_key
libcurses.so:0000000000048ce0 T define_key
libcurses.so:0000000000048bc0 T define_key_sp
nm: 'libtinfo.so': No such file
```

I don't completely understand the issue [from llvm point of view], the ability to detect and use `libtinfo.so` is removed from llvm sources [in 19.0.1 as referred in the above URLs].

```python
        if "+termlib" in self.spec:
            opts.extend(
                (
                    "--with-termlib",
                    "--enable-termcap",
                    "--enable-getcap",
                    "--enable-tcap-names",
                    "--with-versioned-syms",
                )
            )
```

I think the primary configure flag here that results in  this difference is `--with-termlib`  [but then - there are these other configure options also grouped in for  `+~termlib`]

Reply:
I also raised this issue in the PR that introduced the problem:
https://github.com/llvm/llvm-project/pull/92865#issuecomment-2261438018

Basically, the LLVM devs think that because their tests pass, their build env is ok.
But they only test on Darwin/MacOS and apparently on some Linux systms that link libtinfo into libncurses.

But Ubuntu 22.04 and spack (by default) don't do that.

I attest that the `depends_on("ncurses~termlib", when="@19:")` is needed.

Reply:
Resolved by the new push by setting `LLDB_CURSES_LIBS` to `spec["ncurses"].libs`:
```py
if spec.satisfies("@19:"):
    cmake_args.append(define("LLDB_CURSES_LIBS", spec["ncurses"].libs))
```